### PR TITLE
Convert all PF3 buttons to PF4

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

**Comment by garrett** [comment]
I've added this snippet in page.scss for debugging purposes:

```scss
[class*=btn]:not(:hover):not(:focus) {
    filter: drop-shadow(2px 2px red) blur(1px) !important;
}
```

It adds a red dropshadow and blurs the widget slightly when it's not hovered or focused, making all elements with classes containing `btn` obvious.

I'm currently going through Cockpit visually and looking for these buttons. (I've already noticed the close button on the SELinux export dialog is using btn, for example.)

**Comment by KKoukiou** [comment]
> I've added this snippet in page.scss for debugging purposes:
> 
> ```css-scss
> [class*=btn]:not(:hover):not(:focus) {
>     filter: drop-shadow(2px 2px red) blur(1px) !important;
> }
> ```
> 
> It adds a red dropshadow and blurs the widget slightly when it's not hovered or focused, making all elements with classes containing `btn` obvious.
> 
> I'm currently going through Cockpit visually and looking for these buttons. (I've already noticed the close button on the SELinux export dialog is using btn, for example.)

It's using btn-cancel which is a class that does not add-have any styles. it's used only for the tests. That's not a problem.

File: test/verify/check-kdump [hunk 1]
```diff
@@ -60,6 +60,15 @@ class TestKdump(MachineCase):
         b = self.browser
         m = self.machine
 
+        if self.machine.image == "rhel-8-2-distropkg":
+            self.primary_btn_class = '.btn-primary'
```
Comment (martinpitt) [inline test/verify/check-kdump]: This happens several times, feel free to move it into testlib.py's MachineCase.

File: test/verify/check-accounts [hunk 2]
```diff
@@ -343,13 +343,13 @@ class TestAccounts(MachineCase):
 
         # Try an invalid date
         b.set_val("#account-expiration-input", "blah")
-        b.click("#account-expiration .btn-primary")
+        b.click("#account-expiration button:nth-of-type(2)")
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply test/verify/check-accounts]: Right - then I started using that self.{type}_btn_class and format the strings, which made the testing for buttons in ditropkg a bit inconsistent. I can make it consistent if you would like.

File: pkg/dashboard/index.html [hunk 3]
```diff
@@ -23,8 +23,9 @@
     <meta charset="utf-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <link href="../base1/patternfly.css" rel="stylesheet">
-    <link href="../shell/index.css" rel="stylesheet">
+    <link href="dashboard.css" rel="stylesheet">
     <link href="../../static/branding.css" rel="stylesheet">
+    <link rel="stylesheet" href="dashboard.css">
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply pkg/dashboard/index.html]: Good catch!

File: test/verify/check-accounts (hunk 2 above)
[the remark this answers is not in the document]
Comment (martinpitt) [reply test/verify/check-accounts]: Your choice, either is fine with me. I have a slight preference for "primary button" as that's more robust than the position.

**Comment by martinpitt** [comment]
I think "Remove button-sm from several lists" should be squashed into the main commit. Thanks @garrett for the extra fixes!

File: test/verify/check-kdump (hunk 1 above)
Comment (KKoukiou) [reply test/verify/check-kdump]: Done

File: pkg/machines/components/dropdownButtons.jsx [hunk 4]
```diff
@@ -30,40 +31,56 @@ import './dropdownButtons.css';
...
-                {buttons[0].title}
-            </button>
-            <button data-toggle='dropdown' className='btn btn-default dropdown-toggle'>
-                <i className="fa fa-caret-down pf-c-context-selector__toggle-icon" aria-hidden="true" id={caretId} />
-            </button>
-            <ul role='menu' className='dropdown-menu'>
-                {buttonsHtml}
-            </ul>
-        </div>);
+export class DropdownButtons extends React.Component {
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply pkg/machines/components/dropdownButtons.jsx]: Adjusted to use useState

File: test/verify/check-active-pages [hunk 5]
```diff
@@ -89,7 +89,10 @@ class TestActivePages(MachineCase):
         # wait until it's highlighted
         b.wait_present('tr.listing-ct-selected[data-row-id="cockpit1:localhost/system"]')
         # close
-        b.click("#active-pages-dialog button.btn-primary")
+        if m.image == "rhel-8-2-distropkg":
+            b.click("#active-pages-dialog button.btn-primary")
+        else:
+            b.click("#active-pages-dialog button.pf-m-primary")
```
Comment (martinpitt) [inline test/verify/check-active-pages]: This could use the new `primary_btn_class`. But no need to re-push just for this.

**Comment by martinpitt** [comment]
[breaks docker](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-13612-20200303-103429-5fab1adf-debian-stable/log.html)

**Comment by martinpitt** [comment]
Sorry, there are still two failures on [distropkg](https://logs.cockpit-project.org/logs/pull-13612-20200303-165120-6b349207-rhel-8-2-distropkg/log.html). The other failures look like flakes.

**Comment by martinpitt** [comment]
The font styling in [this failure screenshot](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-13612-20200304-102631-e8c98593-centos-8-stream/TestMachinesDBus-testCreate-centos-8-stream-127.0.0.2-2301-FAIL.png) from [this test](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-13612-20200304-102631-e8c98593-centos-8-stream/log.html#182-2) looks really odd. Is that fallout from this PR, or unrelated?

**Comment by KKoukiou** [comment]
> The font styling in [this failure screenshot](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-13612-20200304-102631-e8c98593-centos-8-stream/TestMachinesDBus-testCreate-centos-8-stream-127.0.0.2-2301-FAIL.png) from [this test](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-13612-20200304-102631-e8c98593-centos-8-stream/log.html#182-2) looks really odd. Is that fallout from this PR, or unrelated?

I think it's not that the font is odd - the screenshot just got these in the midway of getting expanded.

**Pull request merged**